.github/workflows/docker-image.yml

```diff
@@ -22,14 +22,17 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
 
-      - name: Get branch or tag name
-        id: ref-name
-        run: echo ::set-output name=ref::${GITHUB_REF#refs/*/}
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/${{ github.repository }}-server
 
       - name: Build and push
         uses: docker/build-push-action@v4
         with:
           context: .
           file: ./misc/server/Dockerfile
-          push: true
-          tags: ghcr.io/${{ github.repository }}-server:${{ steps.ref-name.outputs.ref }}
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
This workflow is not triggered on PRs anyway so I think this is always true?

Reply:
Do we even want to run this workflow on all branches? Wouldn't `master` suffice?

Reply:
> Wouldn't `master` suffice?

I find it helpful for testing. I.e. I can easily point the Protocol Labs Ops team to an image of a new feature developed in a pull request.

As far as I know, public images are free, thus there is no downside to this.

Reply:
```suggestion
          push: true
```

Reply:
> > Wouldn't `master` suffice?
> 
> I find it helpful for testing. I.e. I can easily point the Protocol Labs Ops team to an image of a new feature developed in a pull request.
> 
> As far as I know, public images are free, thus there is no downside to this.

Shall we just add a `workflow_dispatch` trigger then? So it can be triggered for a particular branch?

It seems wasteful to store all of these images if we almost never need them, even if there is no charge associated with it.

Then we can make the condition here: `branch == 'master' || event == 'workflow_dispatch'`.

Reply:
Feel free to merge if you disagree.

Reply:
I guess having to manually trigger the job and then wait for it is also cumbersome. Let's leave it to always push.